VideoAPI/VideoApi.Services/ConsultationService.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using VideoApi.Services.Contracts;
+using VideoApi.Services.Kinly;
+using Task = System.Threading.Tasks.Task;
+
+namespace VideoApi.Services
+{
+    public class ConsultationService : IConsultationService
+    {
+        private readonly IKinlyApiClient _kinlyApiClient;
+        private readonly ILogger<ConsultationService> _logger;
+
+        public ConsultationService(IKinlyApiClient kinlyApiClient, ILogger<ConsultationService> logger)
+        {
+            _kinlyApiClient = kinlyApiClient;
+            _logger = logger;
+        }
+        
+        public async Task<CreateConsultationRoomResponse> CreateConsultationRoomAsync(string virtualCourtRoomId, CreateConsultationRoomParams createConsultationRoomParams)
+        {
+            _logger.LogTrace("Creating a consultation for VirtualCourtRoomId: {virtualCourtRoomId} with prefix {createConsultationRoomParamsPrefix}",
+                virtualCourtRoomId, createConsultationRoomParams.Room_label_prefix);
+
+            CreateConsultationRoomResponse consultationRoomResponse;
+            try
+            {
+                consultationRoomResponse = await _kinlyApiClient.CreateConsultationRoomAsync(virtualCourtRoomId, createConsultationRoomParams);
+            }
+            catch (KinlyApiException e)
+            {
+                _logger.LogError("Unable to create a consultation room for VirtualCourtRoomId: {virtualCourtRoomId}, with exception message: {e}", 
+                    virtualCourtRoomId, e);
+                throw;
```

Comment:
catch block just for logging and then rethrowing is usually frowned upon..

Reply:
same below

Reply:
Any alternative suggestions? If we don’t throw, it’ll swallow the exception and the controller will just carry on with an invalid consultation room or no room at all

Reply:
if you cant handle the exception and dont want to change the exception being thrown (to add more context or a change its type etc)
you should let it throw and be handled and logged elsewhere.